plugin-server/tests/main/ingestion-queues/run-async-handlers-event-pipeline.test.ts

```diff
@@ -53,28 +53,21 @@ describe('runAppsOnEventPipeline()', () => {
     let redis: Redis.Redis
     let closeHub: () => Promise<void>
 
-    beforeEach(() => {
+    beforeEach(async () => {
         // Use fake timers to ensure that we don't need to wait on e.g. retry logic.
         jest.useFakeTimers({ advanceTimers: true })
-    })
-
-    beforeAll(async () => {
-        jest.useFakeTimers({ advanceTimers: true })
         ;[hub, closeHub] = await createHub()
         redis = await hub.redisPool.acquire()
         await hub.postgres.query(PostgresUse.COMMON_WRITE, POSTGRES_DELETE_TABLES_QUERY, null, 'deleteTables') // Need to clear the DB to avoid unique constraint violations on ids
     })
 
-    afterEach(() => {
-        jest.clearAllTimers()
-        jest.useRealTimers()
-        jest.clearAllMocks()
-    })
-
-    afterAll(async () => {
+    afterEach(async () => {
         await hub.redisPool.release(redis)
         await teardownPlugins(hub)
         await closeHub()
+        jest.clearAllTimers()
+        jest.useRealTimers()
+        jest.restoreAllMocks()
     })
```

Comment:
There's some sort of state leakage happening between these tests due to the Kafka producer that is mocked over: https://github.com/PostHog/posthog/blob/e96da9d95d707dca261f71889772b74db3801adc/plugin-server/tests/main/ingestion-queues/run-async-handlers-event-pipeline.test.ts#L109-L111

I couldn't get to the bottom of the actual cause in a reasonable time so I'm not exactly sure what happened. Resetting everything seems to fix it, though.